oe-account-service/op-energy-account-service/src/OpEnergy/BlockTimeStrike/Server/V1/BlockTimeStrikeGuessService.hs

```diff
@@ -258,127 +263,116 @@ getBlockTimeStrikesGuessesPage
+    guessFilter = maybe [] (buildFilter . unFilterRequest . mapFilter) mfilter
     repeatC v = C.yield v >> repeatC v
     sort = maybe Descend (sortOrder . unFilterRequest . id1 . mapFilter) mfilter
       where
         id1 :: FilterRequest BlockTimeStrike BlockTimeStrikeGuessResultPublicFilter -> FilterRequest BlockTimeStrike BlockTimeStrikeGuessResultPublicFilter
         id1 = id -- helping typechecker
-    filters finalFilter linesPerPage =
-      streamEntities
-          (maybe [] (buildFilter . unFilterRequest . mapFilter) mfilter )
-          BlockTimeStrikeGuessId
-          (PageSize ((fromPositive linesPerPage) + 1))
-          sort
-          (Range Nothing Nothing)
-      .| ( C.awaitForever $ \v@(Entity _ guess)-> do
-          C.toProducer $ C.zipSources
-            (repeatC v)
-            (streamEntities
-              ((BlockTimeStrikeId ==. blockTimeStrikeGuessStrike guess):finalFilter)
-              BlockTimeStrikeId
-              (PageSize ((fromPositive linesPerPage) + 1))
-              sort
-              (Range Nothing Nothing)
-            )
-         )
-      .| ( C.awaitForever $ \(guess, strikeE@(Entity strikeId _)) -> do
-           mObserved <- lift $ selectFirst [ BlockTimeStrikeObservedStrike ==. strikeId ][]
-           case maybe Nothing (blockTimeStrikeGuessResultPublicFilterClass . fst . unFilterRequest) mfilter of
-             Nothing -> C.yield (guess, strikeE, mObserved) -- don't care about existence of observed data
-             Just BlockTimeStrikeFilterClassGuessable-> case mObserved of
-               Nothing -> C.yield (guess, strikeE, mObserved) -- strike have not been observed and finalFilter should ensure, that it is in the future with proper guess threshold
-               _ -> return () -- otherwise, block haven't matched the criteria and should be ignored
-             Just BlockTimeStrikeFilterClassOutcomeUnknown-> case mObserved of
-               Nothing-> C.yield (guess, strikeE, mObserved) -- haven't been observed
-               _ -> return ()
-             Just BlockTimeStrikeFilterClassOutcomeKnown-> case mObserved of
-               Just _ -> C.yield (guess, strikeE, mObserved) -- had been observed
-               _ -> return ()
-         )
-      .| ( C.awaitForever $ \v@( Entity _ guess, _, _ )-> do
-          C.toProducer $ C.zipSources
-            (repeatC v)
-            ( streamEntities
-              (( PersonId ==. blockTimeStrikeGuessPerson guess ):(maybe [] (buildFilter . unFilterRequest . mapFilter) mfilter ))
-              PersonId
-              (PageSize ((fromPositive linesPerPage) + 1))
-              sort
-              (Range Nothing Nothing)
-            )
-         )
+
+    streamGuessStrikeObservedResultAndOwner finalFilter linesPerPage
+      = fetchPerGuessBlockTimeStrikeAndZipThem
+      .| possiblyFetchObservedResultAndZipWithGuessAndStrike
+      .| fetchPerGuessOwnerAndZipObservedResultGuessAndStrike
+      .| buildBlockTimeStrikeGuessResultPublic
```

Comment:
While refactoring to use existing transducer combinators, I would likely move the combinators into this pipeline. For instance,  `streamGuessStrikeObservedResultAndOwner` might become:
```suggestion
    streamGuessStrikeObservedResultAndOwner finalFilter linesPerPage
      = zipConcatMapMC fetchPerGuessBlockTimeStrike
      .| C.mapMC fetchObservedResult
      .| zipConcatMapMC fetchPerGuessOwner
      .| C.mapC buildBlockTimeStrikeGuessResultPublic
```

Here I have exploited the fact that this idiom is repeated:
```
-- | Map over the items of a Conduit, tupling each result
-- with its preimage.
zipConcatMapMC :: (a -> ConduitT () b m ()) -> Conduit a (a,b) m ()
```

Reply:
implemented this `zipConcatMC`: https://github.com/op-energy-foundation/op-energy/pull/41/files#diff-b78c7abab98ec348306ac8ef2f08787345923216c38370a988cd8212d4317ab0